orangecontrib/prototypes/widgets/owfeaturestatistics.py

```diff
@@ -0,0 +1,595 @@
+        # crucial, since otherwise the `ResizeToContents` section resize mode
+        # would call `sizeHint` on every single row in the data before first
+        # render. However this, this cannot be used here, since this only
+        # appears to work properly when the widget is actually shown. When the
+        # widget is not shown, size `sizeHint` is called on every row.
+        hheader.setResizeContentsPrecision(5)
+        # Set a nice default size so that headers have some space around titles
+        hheader.setDefaultSectionSize(120)
+        # Set individual column behaviour in `set_data` since the logical
+        # indices must be valid in the model, which requires data.
+        hheader.setSectionResizeMode(QHeaderView.Interactive)
+
+        vheader = self.view.verticalHeader()
+        vheader.setVisible(False)
+        vheader.setSectionResizeMode(QHeaderView.Fixed)
+
+        def bind_histogram_aspect_ratio(logical_index, _, new_size):
+            """Force the horizontal and vertical header to maintain the defined
+            aspect ratio specified for the histogram."""
+            if logical_index is not self.model.Columns.DISTRIBUTION.index:
+                return
+            ratio_width, ratio_height = self.HISTOGRAM_ASPECT_RATIO
+            unit_width = new_size / ratio_width
+            new_height = unit_width * ratio_height
+            effective_height = max(new_height, self.MINIMUM_HISTOGRAM_HEIGHT)
+            vheader.setDefaultSectionSize(effective_height)
+
+        def keep_row_centered(logical_index, old_size, new_size):
+            # We only need to call this once per resize
+            if logical_index is not self.model.Columns.DISTRIBUTION.index:
+                return
+            top_row = self.view.indexAt(self.view.rect().topLeft()).row()
+            bottom_row = self.view.indexAt(self.view.rect().bottomLeft()).row()
+            middle_row = top_row + (bottom_row - top_row) // 2
+            self.view.scrollTo(self.model.index(middle_row, 0), QTableView.PositionAtCenter)
+
+        hheader.sectionResized.connect(bind_histogram_aspect_ratio)
+        hheader.sectionResized.connect(keep_row_centered)
+
+        self.distribution_delegate = DistributionDelegate()
+        self.view.setItemDelegate(self.distribution_delegate)
+
+        self.mainArea.layout().addWidget(self.view)
+
+    def sizeHint(self):
+        return QSize(900, 500)
+
+    def _filter_table_variables(self):
+        regex = QRegExp(self.filter_string)
+        # If the user explicitly types different cases, we assume they know
+        # what they are searching for and account for letter case in filter
+        different_case = (
+            any(c.islower() for c in self.filter_string) and
+            any(c.isupper() for c in self.filter_string)
+        )
+        if not different_case:
+            regex.setCaseSensitivity(Qt.CaseInsensitive)
+
+        self.proxy_model.setFilterRegExp(regex)
```

Comment:
Looks like a remnant since https://github.com/biolab/orange3-prototypes/pull/72/commits/2e440239cf6b98db1968d746b38f0f3e4386e074. `AbstractTableModel` unfortunately does not do filtering. So you should probably adapt/remove this.

But I think filtering is nice to have. @pavlin-policar What are your thoughts on it? Would you be willing to amend `AbstractSortTableModel` with filtering capabilities, or wrap `FeatureStatisticsTableModel` in a `QSortFilterProxyModel` just for its filtering?

Reply:
My bad, I must have missed it when converting over to the new `AbstractTableModel`.

I think filtering is a very important feature, since the GEO datasets can have thousands of features and looking for features by scrolling can get tedious, very fast. I will look into it.